frontend/src/components/Playback/SubtitleTrack.vue

```diff
@@ -0,0 +1,100 @@
+<template>
+  <div
+    class="uno-absolute uno-bottom-0 uno-left-0 uno-w-full uno-text-center">
+    <span
+      class="uno-inline-block uno-pb-10px uno-color-white"
+      :class="{ 'stroked': subtitleSettings.state.stroke }"
+      :style="subtitleStyle">
+      <template v-if="preview">
+        {{ $t('subtitlePreviewText') }}
+      </template>
+      <JSafeHtml
+        v-else-if="currentSubtitle !== undefined"
+        :html="currentSubtitle.sub.text" />
+    </span>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed, type StyleValue } from 'vue';
+import { subtitleSettings } from '@/store/client-settings/subtitle-settings';
+import { DEFAULT_TYPOGRAPHY } from '@/store';
+import { playerElement } from '@/store/player-element';
+import { isNil } from '@/utils/validation';
+import type { ParsedSubtitleTrack, Dialogue } from '@/plugins/workers/generic/subtitles';
+import { playbackManager } from '@/store/playback-manager';
+
+const { preview } = defineProps<{
+  /**
+   * Whether the subtitle track is in preview mode.
+   */
+  preview?: boolean;
+}>();
+
+/**
+ * Update the current subtitle based on the current time of the media element.
+ *
+ * Loops in the first run (we can't assume that the first run will appear at index 0,
+ * since the user can seek to any position) when 'previous' is undefined and then relies in previous
+ * to find the next one
+ */
+const predicate = (d: Dialogue) => d.start <= playbackManager.currentTime && d.end >= playbackManager.currentTime;
+const findSubtitle = (dialogue: ParsedSubtitleTrack['dialogue'], start = 0) =>
+  dialogue.slice(start).findIndex(d => predicate(d));
+
+const dialogue = computed(() => playerElement.currentExternalSubtitleTrack?.parsed?.dialogue);
+const currentSubtitle = computed<{ index: number; sub: Dialogue } | undefined>((previous) => {
+  if (!isNil(dialogue.value)) {
+    const hasPrevious = !isNil(previous);
+    const isNext = hasPrevious && predicate(dialogue.value[previous.index + 1]);
+    const isCurrent = hasPrevious && predicate(dialogue.value[previous.index]);
+
+    if (isCurrent) {
+      return previous;
+    } else {
+      const newIndex = isNext ? previous.index + 1 : findSubtitle(dialogue.value);
+
+      return { index: newIndex, sub: dialogue.value[newIndex] };
+    }
```

Comment:
@seanmcbroom This is the day of things that we think we did but we didn't 🤦‍♂️.

 If you check `findSubtitle`, it returns `undefined` instead of -1 when no element is found, so your condition as is will never be truthy (instead we should check for nil, good catch!).

The question I have now is whether what do you think is more correct: return undefined or return the previous good known value?

Reply:
If no subtitle is found it should return undefined, since there can be intentional gaps without subtitles.

Reply:
I also found out thanks to your comment that I forgot to pass the last found index to `findSubtitle`!